apps/wizard/app_pages/producer_analytics/data_io.py

```diff
@@ -113,8 +115,68 @@ def get_chart_views_from_bq(
     return cast(pd.DataFrame, df_views)
 
 
-@st.cache_data(show_spinner=False)
+@st_cache_data(custom_text="Getting producers per chart from database...")
 def get_producers_per_chart(excluded_steps) -> pd.DataFrame:
+    """Get producers per chart from the DB.
+
+    Query traces back the details of the chart:
+    chart -> variable -> producer.
+
+    Additionally, it incorporates the dataset URI and the dataset name of the variable in use by the chart. This will help us exclude certain steps (e.g. population).
+
+    TODO: add support to exclude steps.
+    """
+    query = """WITH t_base AS (
+	SELECT
+		cd.chartId chart_id,
+		cf.slug chart_slug,
+		JSON_EXTRACT(cf.full, '$.isPublished') is_published,
+		cd.variableId variable_id,
+		v.name variable_name,
+		d.id dataset_id,
+		d.name dataset_name,
+		d.catalogPath dataset_uri,
+		ov.originId origin_id,
+		o.title origin_name,
+        o.urlMain origin_url,
+		o.producer producer
+	FROM chart_dimensions cd
+	LEFT JOIN charts c ON c.id = cd.chartId
+	LEFT JOIN chart_configs cf ON cf.id = c.configId
+	LEFT JOIN origins_variables ov ON ov.variableId = cd.variableId
+	LEFT JOIN origins o ON o.id = ov.originId
+	LEFT JOIN variables v ON cd.variableId = v.id
+	LEFT JOIN datasets d ON d.id = v.datasetId
+)
+SELECT * FROM t_base
+WHERE origin_id IS NOT NULL
+AND is_published = true;
```

Comment:
Nice! I assumed such a query would not be possible, but I was not aware of `origins_variables`. Thanks for figuring this out.